include/nbl/builtin/hlsl/bxdf/ndf.hlsl

```diff
@@ -0,0 +1,339 @@
+// Copyright (C) 2018-2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_BXDF_NDF_INCLUDED_
+#define _NBL_BUILTIN_HLSL_BXDF_NDF_INCLUDED_
+
+#include "nbl/builtin/hlsl/limits.hlsl"
+#include "nbl/builtin/hlsl/bxdf/common.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace bxdf
+{
+namespace ndf
+{
```

Comment:
the NDF "fixes" the Lambda and Beta functions so it makes sense to have an `ndf(this_t::query_type)` (instead of just `operator(this_t::query_type)`) and `lambda()` methods instead of having the lambda buried somewhere else in anothre header

https://github.com/Devsh-Graphics-Programming/Nabla/pull/811#discussion_r2038366946

Reply:
actualy `ndf` should be called `D` to be consistent with that PBR papers (Cook Torrance)

then vndf is `DG1`

Reply:
So to recap:
- D (the ndf)
- Lambda
- Beta
- DG1
- Generate (take a `localV` not an entire interaction)
- optional: generate overload which can output microfacet cache (not sure all NDFs will be able to do this